### ISSUE TITLE
Details of Codes

### ISSUE THREAD

**Issue by axhiao** [junkangwu/BSL#2]
Hi, based on your codes in the Documentation as below:
```python
pos_logits = torch.exp(y_pred[:, 0] / self.temperature)
neg_logits = torch.exp(y_pred[:, 1:] / self.temperature)
neg_logits = neg_logits.sum(dim=-1)
neg_logits = torch.pow(neg_logits, self.temperature_2)
loss = - torch.log(pos_logits / neg_logits).mean()
```
Are these key codes for BSL? If so, how do they differ from the implementation of softmax loss? If `temperature_2=1` in the code above, it seems to me that this is simply an implementation of SL, as illustrated in the image below. Thank you!

<img width="736" alt="image" src="https://github.com/user-attachments/assets/17c4302a-f251-4882-a6e1-d1bfb7365bfd" />

**Comment by junkangwu** [comment]
Yes, when `temperature_2 = 1`, it degrades to SL, but it is precisely because of `temperature_2` that BSL can achieve robust learning for positive samples and significantly improve performance on existing tasks.

This is also the core of this work: a very simple implementation with significant results and clear motivation.
